.github/workflows/jira.yaml

```diff
@@ -0,0 +1,44 @@
+# this workflow requires to provide JIRA webhook URL via JIRA_URL GitHub Secret
+# read more: https://support.atlassian.com/cloud-automation/docs/jira-automation-triggers/#Automationtriggers-Incomingwebhook
+# original code source: https://github.com/beliaev-maksim/github-to-jira-automation
+
+name: Issues to JIRA
+
+on:
+  workflow_call:
+
+jobs:
+  update:
+    name: Update Issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create JIRA ticket
+        env:
+          # put into env vars to properly escape special bash chars
+          ISSUE_TITLE: ${{ github.event.issue.title }}
+          ISSUE_DESCRIPTION: ${{ github.event.issue.body }}
+        run: |
+          # Determine Bug/Enhancement
+          if ${{ contains(github.event.*.labels.*.name, 'Bug') }}; then
+            type=bug
+          else
+            type=story
+          fi
+
+          # send JIRA request
+
+          # Canonical discourages discussions in JIRA on public project issues.
+          # It is NOT recommended to put description in JIRA in order to
+          # push conversation to GitHub. If strongly required,
+          # replace '--arg body ""' with '--arg body "$ISSUE_DESCRIPTION"'
+
+          data=$( jq -n \
+                  --arg title "$ISSUE_TITLE" \
+                  --arg url '${{ github.event.issue.html_url }}' \
+                  --arg submitter '${{ github.event.issue.user.login }}' \
+                  --arg body "" \
+                  --arg type "$type" \
+                  --arg action '${{ github.event.action }}' \
+                  '{title: $title, url: $url, submitter: $submitter, body: $body, type: $type, action: $action}' )
+
+          curl -X POST -H 'Content-type: application/json' --data "${data}" "${{ secrets.JIRA_URL }}"
```

Comment:
I'm surprised, isn't there any API key secret or authentication method required to post to Jira ?
How does this work ?

Reply:
The API key is part of the URL, that's why I created it as a secret. It is generated as a webhook URL by Jira in the Automation section.

Reply:
Also thinking twice about it, in my opinion it would be better to "hardcode" the hostname, and eventually the path, and just store the token as secret.
It won't be very reusable but I don't like mixing real secrets with other data